File: 2019W11/preppindataw11.R
library(dplyr)
library(readr)

jdata <- read_csv("E:/PD - JSON DATA Stock data.csv",
                  col_types = cols(JSON_Name = col_character(),
                                   JSON_ValueString = col_character())) %>%
  na.omit() 

final <- jdata %>%
  mutate('Category'= lapply(strsplit(`JSON_Name`, '\\.'), function(x) x[4])) %>%
  filter(`Category` != 'meta') %>%
  mutate('Row'= lapply(lapply(strsplit(`JSON_Name`, '\\.'),rev), function(x) x[1])) %>%
  mutate('DataType'= lapply(lapply(strsplit(`JSON_Name`, '\\.'),rev), function(x) x[2])) %>%
  select(c('Row','DataType','JSON_ValueString')) %>%
  spread('DataType', 'JSON_ValueString') %>%
  mutate('Date'=as.POSIXct(as.integer(`timestamp`), origin="1970-01-01",tz = "GMT")) %>%
  select(c('Date','volume','high','low','adjclose','close','open','Row'))

View(final)

File: 2019/2019W11/PreppindataW11.py
import pandas as pd
from datetime import datetime

jdata = pd.read_csv("E:\PD - JSON DATA Stock data.csv").dropna()
jdata['Row']=jdata.apply(lambda x: [x for x in x['JSON_Name'].split('.')][::-1][0],axis=1)
jdata['DataType']=jdata.apply(lambda x: [x for x in x['JSON_Name'].split('.')][::-1][1] ,axis=1)
jdata['Category']=jdata.apply(lambda x: [x for x in x['JSON_Name'].split('.')][3] ,axis=1)

final = jdata[jdata['Category'] != 'meta'][['Row','DataType','JSON_ValueString']].copy()
final = final.pivot_table(index = ['Row'],
                          columns='DataType',
                          values='JSON_ValueString',
                          aggfunc='max')
final.reset_index(inplace = True)
final['Date']=final.apply(lambda x: datetime.utcfromtimestamp(int(x['timestamp'])).strftime('%Y-%m-%d %H:%M:%S') ,axis=1)
final=final[['Date','volume','high','low','adjclose','close','open','Row']]
